### ISSUE TITLE
oa2df doesn't seem to work with lists of different length

### ISSUE THREAD

**Issue by inesjuliak** [ropensci/openalexR#111]
Hi,

there seems to be a problem with oa2df  when oa_requests results in lists of different length. In this case, oa2df does not seem to be able to convert into dataframes. For example:

`ID = oa_request(query_url=
                  "https://api.openalex.org/authors?filter=display_name.search:John%Snow")
ID= oa2df(ID, "authors")`

gives me the error message:

> error in (function (..., deparse.level = 1, make.row.names = TRUE, stringsAsFactors = FALSE,  : 
>   numbers of columns of arguments do not match

Some profiles are missing the "relevance_score", which leads to the difference in length in the lists. is there an easy fix/workaround for this, other than filtering out the relevance_score in advance?

Thank you in advance!

**Comment by trangdata** [comment]
Hi @inesjuliak this has been fixed in #108. Could you reinstall the package from GitHub and try again? 

``` r
install.packages("remotes")
remotes::install_github("ropensci/openalexR")
``` 

``` r
library(openalexR)
oa_fetch(
  "authors",
  display_name.search = "John Snow",
  verbose = TRUE
)
#> Requesting url: https://api.openalex.org/authors?filter=display_name.search%3AJohn%20Snow
#> Getting 1 page of results with a total of 79 records...
#> # A tibble: 79 × 16
#>    id         display_name display_name_alterna…¹ relevance_score ids      orcid
#>    <chr>      <chr>        <lgl>                            <dbl> <list>   <lgl>
#>  1 https://o… John Snow    NA                               2169. <tibble> NA   
#>  2 https://o… John Snow    NA                               1754. <tibble> NA   
#>  3 https://o… John Snow    NA                               1386. <tibble> NA   
#>  4 https://o… John Snow    NA                                868. <tibble> NA   
#>  5 https://o… John Snow    NA                                726. <tibble> NA   
#>  6 https://o… John Snow    NA                                684. <tibble> NA   
#>  7 https://o… John Snow    NA                                673. <tibble> NA   
#>  8 https://o… John Snow    NA                                662. <tibble> NA   
#>  9 https://o… John Snow    NA                                494. <tibble> NA   
#> 10 https://o… John Snow    NA                                471. <tibble> NA   
#> # ℹ 69 more rows
#> # ℹ abbreviated name: ¹​display_name_alternatives
#> # ℹ 10 more variables: works_count <int>, cited_by_count <int>,
#> #   counts_by_year <list>, affiliation_display_name <chr>,
#> #   affiliation_id <chr>, affiliation_ror <chr>,
#> #   affiliation_country_code <chr>, affiliation_type <chr>, x_concepts <list>,
#> #   works_api_url <chr>
```

<sup>Created on 2023-05-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by inesjuliak** [comment]
> Hi @inesjuliak this has been fixed in #108. Could you reinstall the package from GitHub and try again?
> 
> ```r
> install.packages("remotes")
> remotes::install_github("ropensci/openalexR")
> ```
> 
> ```r
> library(openalexR)
> oa_fetch(
>   "authors",
>   display_name.search = "John Snow",
>   verbose = TRUE
> )
> #> Requesting url: https://api.openalex.org/authors?filter=display_name.search%3AJohn%20Snow
> #> Getting 1 page of results with a total of 79 records...
> #> # A tibble: 79 × 16
> #>    id         display_name display_name_alterna…¹ relevance_score ids      orcid
> #>    <chr>      <chr>        <lgl>                            <dbl> <list>   <lgl>
> #>  1 https://o… John Snow    NA                               2169. <tibble> NA   
> #>  2 https://o… John Snow    NA                               1754. <tibble> NA   
> #>  3 https://o… John Snow    NA                               1386. <tibble> NA   
> #>  4 https://o… John Snow    NA                                868. <tibble> NA   
> #>  5 https://o… John Snow    NA                                726. <tibble> NA   
> #>  6 https://o… John Snow    NA                                684. <tibble> NA   
> #>  7 https://o… John Snow    NA                                673. <tibble> NA   
> #>  8 https://o… John Snow    NA                                662. <tibble> NA   
> #>  9 https://o… John Snow    NA                                494. <tibble> NA   
> #> 10 https://o… John Snow    NA                                471. <tibble> NA   
> #> # ℹ 69 more rows
> #> # ℹ abbreviated name: ¹​display_name_alternatives
> #> # ℹ 10 more variables: works_count <int>, cited_by_count <int>,
> #> #   counts_by_year <list>, affiliation_display_name <chr>,
> #> #   affiliation_id <chr>, affiliation_ror <chr>,
> #> #   affiliation_country_code <chr>, affiliation_type <chr>, x_concepts <list>,
> #> #   works_api_url <chr>
> ```
> 
> Created on 2023-05-24 with [reprex v2.0.2](https://reprex.tidyverse.org)

Ah thank you for the quick reply & sorry, I somehow overlooked #108, works fine now after update
